### PR TITLE
feat(web/applications): add page to edit transcript field (boas-1086)

### PR DESCRIPTION
**Pull request by lutamKC**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/views/applications/case-documentation/properties/tab-details.component.njk [hunk 1]
```diff
@@ -1,7 +1,16 @@
 {% from "govuk/components/summary-list/macro.njk" import govukSummaryList %}
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
 
 {% macro documentationDetailsTab(documentationFile, caseId, folderId) %}
 
+	{% if documentationFile.mime === 'image/png' %}
```
Comment (francescobrenci-AR) [inline apps/web/src/server/views/applications/case-documentation/properties/tab-details.component.njk:6]: this is the wrong MIMe, should be Video/audio/HTML

File: apps/api/src/server/applications/application/documents/document.controller.js [hunk 2]
```diff
@@ -489,6 +489,24 @@ export const storeDocumentVersion = async (request, response) => {
...
+		const transcriptDocument = await documentRepository.getByReferenceRelatedToCaseId(
+			transcriptReference,
+			+caseId
+		);
+
+		if (!transcriptDocument) {
+			const transcriptErrorMsg = `Transcript document not found: reference ${transcriptReference}`;
+			response.status(404).send({ errors: { transcript: transcriptErrorMsg } });
+
+			throw new BackOfficeAppError(transcriptErrorMsg, 404);
```
Comment (HadrianHughesKC) [inline apps/api/src/server/applications/application/documents/document.controller.js:504]:
```suggestion
			throw new BackOfficeAppError(transcriptErrorMsg, 404);
```

If you throw a `BackOfficeAppError` it should implicitly respond with a 404 and the error message you give it, so you don't need to do it explicitly.

File: apps/api/src/server/applications/application/documents/document.controller.js [hunk 3]
```diff
@@ -489,6 +489,24 @@ export const storeDocumentVersion = async (request, response) => {
 		throw new BackOfficeAppError(`Document not found: guid ${guid}`, 404);
 	}
 
+	if (documentVersionMetadataBody.transcript) {
+		const transcriptReference = documentVersionMetadataBody.transcript;
+
+		const transcriptDocument = await documentRepository.getByReferenceRelatedToCaseId(
+			transcriptReference,
+			+caseId
```
Comment (HadrianHughesKC) [inline apps/api/src/server/applications/application/documents/document.controller.js:497]:
```suggestion
			Number(caseId)
```

Better to be explicit.

**Review by HadrianHughesKC — APPROVED** [review]
Looks good. I've left a couple of suggestions but nothing that should block.

**Pull request merged**